ssc/libssc.c

```diff
@@ -796,19 +1232,26 @@ static PetscErrorCode PCPatchComputeOperator(PC pc, Mat mat, PetscInt which)
 
     ierr = PetscSectionGetDof(patch->cellCounts, which, &ncell); CHKERRQ(ierr);
     ierr = PetscSectionGetOffset(patch->cellCounts, which, &offset); CHKERRQ(ierr);
+    if ( ncell <= 0 ) {
+        ierr = PetscLogEventEnd(PC_Patch_ComputeOp, pc, 0, 0, 0); CHKERRQ(ierr);
+        PetscFunctionReturn(0);
+    }
     PetscStackPush("PCPatch user callback");
-    ierr = patch->usercomputeop(pc, mat, ncell, cellsArray + offset, ncell*patch->nodesPerCell, dofsArray + offset*patch->nodesPerCell, patch->usercomputectx); CHKERRQ(ierr);
+    ierr = patch->usercomputeop(pc, mat, ncell, cellsArray + offset, ncell*patch->totalDofsPerCell, dofsArray + offset*patch->totalDofsPerCell, patch->usercomputectx); CHKERRQ(ierr);
     PetscStackPop;
     ierr = ISRestoreIndices(patch->dofs, &dofsArray); CHKERRQ(ierr);
     ierr = ISRestoreIndices(patch->cells, &cellsArray); CHKERRQ(ierr);
+
     /* Apply boundary conditions.  Could also do this through the local_to_patch guy. */
+    if (patch->multiplicative) {
+        ierr = MatCopy(mat, multMat, DIFFERENT_NONZERO_PATTERN); CHKERRQ(ierr);
```

Comment:
This should be `SAME_NONZERO_PATTERN` now that preallocation is correct, I believe.

Reply:
Check if we should use `MatDuplicate`?  Probably not.